outlines/blaze.md

```diff
@@ -0,0 +1,56 @@
+# Blaze
+
+1. Introduction to Blaze -- the tracker-backed handlebars like templating syntax
+  1. Example of spacebars syntax, data context + helpers
+  2. Data contexts and access - name lookup (`.`, `this`, `..`, `[0]`), and null values
+  3. Helpers, arguments, options
+  4. Inclusion of other templates + arguments, passing data contexts
+  5. Helpers inside tags -- returning strings and objects, `checked={{isChecked}}`
+  6. Nested helpers + sub expressions
+  7. Block helpers (you can create them with templates, see 10. below)
+  8. Safestrings and `{{{`
+  9. Builtin block helpers
+    1. `{{#if/unless}}`
+    2. `{{#with}}`
+    3. `{{#each}}`
+    3. `{{#let}}`
+  10. Comments
+  11. Strictness
+  12. Escaping
+2. Creating reusable "pure" components with Blaze / best practice (a lot of this is repeating @sanjo's boilerplate)
+  1. Validating data context fits a schema
+  2. Always set data contexts to `{name: doc}` rather than just `doc`. Always set a d.c on an inclusion.
+  3. Use `{{#each .. in .. }}` to achieve the above
+  4. Use the template instance as a component -- adding a `{{template}}` helper to access it
+  5. Use a (named / scoped on _id if possible) reactive dict for instance state
```

Comment:
Hm, I really think we should not make it be a part of template instance just so. We have too many places where people can store then stuff, into `state`, into template instance, into data context. Too many choices then mean people will be confused where to store it, and some people will use something, other something else so things will be less interoperable.

I really think that `{{instance.counter}}` is much nicer than `{{instance.state.get 'counter'}}`. Moreover, it allows you to have multiple stuff attached to `instance`. Like computer field, or simply a method/function. And you can then call all in the same way, with `instance.something`. Otherwise you have some stuff in `state`, some stuff in data context, some stuff as a method, some stuff as a computed field.

I really think this is bad design. There is no need for this. While my other comments are just  comments, here I really think strongly that the best thing is just to encourage use of template instances without any need for extra sub-namespacing into `state`.